R/pmcmc_parameters.R

```diff
@@ -46,6 +46,9 @@ pmcmc_parameter <- function(name, initial, min = -Inf, max = Inf,
   if (initial > max) {
     stop(sprintf("'initial' must be <= 'max' (%s)", max))
   }
+  if (integer & round(initial) != initial) {
+    stop(sprintf("'initial' must be an integer but was %s", initial))
+  }
```

Comment:
Can we route any of this through your new `pmcmc_check_initial_values` function?